File: csharp/DVSClient/DVSClient/server/address/suggestions/RestApiSuggestionsFormatSuggestion.cs
using DVSClient.Server.Address.Format;
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Suggestions
{
    public class RestApiSuggestionsFormatSuggestion
    {
        [JsonProperty("global_address_key")]
        public string? GlobalAddressKey { get; set; }

        [JsonProperty("address")]
        public RestApiFormatAddress? Address { get; set; }

        [JsonProperty("addresses_formatted")]
        public IEnumerable<RestApiAddressFormatted>? AddressesFormatted { get; set; }

        [JsonProperty("components")]
        public RestApiAddressFormatComponents? Components { get; set; }

        [JsonProperty("metadata")]
        public RestApiAddressFormatMetadata? Metadata { get; set; }
    }
}

File: typescript/src/server/address/suggestions/restApiSuggestionsFormatSuggestion.ts
import { RestApiAddressFormatComponents } from "../format/restApiAddressFormatComponents";
import { RestApiAddressFormatMetadata } from "../format/restApiAddressFormatMetadata";
import { RestApiAddressFormatted } from "../format/restApiAddressFormatted";
import { RestApiFormatAddress } from "../format/restApiFormatAddress";

export type RestApiSuggestionsFormatSuggestion = {
    global_address_key?: string;
    address?: RestApiFormatAddress;
    addresses_formatted?: RestApiAddressFormatted[];
    components?: RestApiAddressFormatComponents;
    metadata?: RestApiAddressFormatMetadata;
};